tsdb/db.go

```diff
@@ -638,6 +638,21 @@ func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs
 		return nil, err
 	}
 
+	defer func() {
+		// Close files if startup fails somewhere
+		if err != nil {
+			if err = db.head.Close(); err != nil {
```

Comment:
You should NOT override err

Reply:
Oh thanks for pointing that out! :open_mouth:

Reply:
It is because this function is using named returns